### PR TITLE
WinUI's UniformGridLayout implementation for ItemsRepeater

### PR DESCRIPTION
**Pull request by carldebilly**
Implements #5611 

# Feature
Add support for `UniformGridLayout`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- ~~[ ] Docs have been added/updated which fit [documentation template]~~(https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.macOS.cs [hunk 1]
```diff
@@ -105,10 +105,6 @@ internal Windows.Foundation.Point GetPosition(Point position, global::Windows.UI
 #endif
 		}
 
-#if DEBUG
-		public string ShowLocalVisualTree(int fromHeight) => AppKit.UIViewExtensions.ShowLocalVisualTree(this, fromHeight);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/UIElement.macOS.cs:109]: why are you removing this, was it failing to build?

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/UniformGridLayout/UniformGridLayout.cs [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class UniformGridLayout : IFlowLayoutAlgorithmDelegates
+	{
+		#region IGridLayout
+
+		public UniformGridLayout()
+		{
+			this.RegisterDisposablePropertyChangedCallback((i, s, e) => OnPropertyChanged(e));
+			//__RP_Marker_ClassById(RuntimeProfiler.ProfId_UniformGridLayout);
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/UniformGridLayout/UniformGridLayout.cs:20]: remove

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/UniformGridLayout/UniformGridLayout.cs [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+			if (state is { })
+			{
+				gridState = GetAsGridState(state);
+			}
+
+			if (!(gridState is { }))
+			{
+				if (state is { })
+				{
+					//throw hresult_error(E_FAIL, "LayoutState must derive from UniformGridLayoutState.");
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/UniformGridLayout/UniformGridLayout.cs:41]: remove

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**